gamla/functional_generic.py

```diff
@@ -395,10 +457,20 @@ async def apply_spec_async(*args, **kwargs):
 
 
 def bifurcate(*funcs):
-    """Serially runs each function on tee'd copies of `input_generator`."""
+    """Serially run each function on tee'd copies of a sequence.
+    If the sequence is a generator, it is exhausted only once.
+    >>> f = bifurcate(sum, gamla.count)
+    >>> seq = map(gamla.identity, [1, 2, 3, 4, 5])
+    >>> f(seq)
+    (15, 5)
+    """
     return compose_left(iter, lambda it: itertools.tee(it, len(funcs)), stack(funcs))
 
 
+#:  Average of an iterable.
```

Comment:
might be useful to say how it handles an empty one (just returns 0)

Reply:
not sure this behavior is the correct global one. e.g. numpy returns `nan` in this case.

Reply:
we could throw a ValueError